commands/dns.go

```diff
@@ -77,40 +77,77 @@ func (cmd *DNS) RemoveHostFilter(ipAddr string) {
-		exec.Command("bash", "-c", "echo \"nameserver "+bridgeIP+"\" | sudo tee /etc/resolver/vm").Run()
-		exec.Command("sudo", "route", "-n", "delete", "-net", "172.17.0.0").Run()
-		util.StreamCommand(exec.Command("sudo", "route", "-n", "add", "172.17.0.0/16", machineIP))
-
-		if _, err := os.Stat("/usr/sbin/discoveryutil"); err == nil {
-			// Put this here for people running OS X 10.10.0 to 10.10.3 (oy vey.)
-			cmd.out.Verbose.Println("Restarting discoveryutil to flush DNS caches")
-			util.StreamCommand(exec.Command("sudo", "launchctl", "unload", "-w", "/System/Library/LaunchDaemons/com.apple.discoveryd.plist"))
-			util.StreamCommand(exec.Command("sudo", "launchctl", "load", "-w", "/System/Library/LaunchDaemons/com.apple.discoveryd.plist"))
-		} else {
-			// Reset DNS cache. We have seen this suddenly make /etc/resolver/vm work.
-			cmd.out.Verbose.Println("Restarting mDNSResponder to flush DNS caches")
-			util.StreamCommand(exec.Command("sudo", "killall", "-HUP", "mDNSResponder"))
+
+	if machine.IsXhyve() {
+		cmd.RemoveHostFilter(machine.GetIP())
+	}
+	exec.Command("sudo", "mkdir", "-p", "/etc/resolver").Run()
+	exec.Command("bash", "-c", "echo \"nameserver "+bridgeIP+"\" | sudo tee /etc/resolver/vm").Run()
+	exec.Command("sudo", "route", "-n", "delete", "-net", "172.17.0.0").Run()
+	util.StreamCommand(exec.Command("sudo", "route", "-n", "add", "172.17.0.0/16", machineIP))
+	if _, err := os.Stat("/usr/sbin/discoveryutil"); err == nil {
+		// Put this here for people running OS X 10.10.0 to 10.10.3 (oy vey.)
+		cmd.out.Verbose.Println("Restarting discoveryutil to flush DNS caches")
+		util.StreamCommand(exec.Command("sudo", "launchctl", "unload", "-w", "/System/Library/LaunchDaemons/com.apple.discoveryd.plist"))
+		util.StreamCommand(exec.Command("sudo", "launchctl", "load", "-w", "/System/Library/LaunchDaemons/com.apple.discoveryd.plist"))
+	} else {
+		// Reset DNS cache. We have seen this suddenly make /etc/resolver/vm work.
+		cmd.out.Verbose.Println("Restarting mDNSResponder to flush DNS caches")
+		util.StreamCommand(exec.Command("sudo", "killall", "-HUP", "mDNSResponder"))
+	}
+}
+
+// ConfigureLinux configures network routing and DNS resolution
+func (cmd *DNS) ConfigureLinux() error {
+	bridgeIP, err := util.GetBridgeIP()
+	if err != nil {
+		return err
+	}
+
+	// Is NetworkManager in use
+	if _, err := os.Stat("/etc/NetworkManager/dnsmasq.d"); err == nil {
+		// Install for NetworkManager/dnsmasq connection to dnsdock
+		util.StreamCommand(exec.Command("bash", "-c", fmt.Sprintf("echo 'server=/vm/%s' | sudo tee /etc/NetworkManager/dnsmasq.d/dnsdock.conf", bridgeIP)))
+
+		// Restart NetworkManager if it is running
+		if err := exec.Command("systemctl", "is-active", "NetworkManager").Run(); err != nil {
+			util.StreamCommand(exec.Command("sudo", "systemctl", "restart", "NetworkManager"))
 		}
 	}
+
+	// Is libnss-resolver in use
+	if _, err := os.Stat("/etc/resolver"); err == nil {
```

Comment:
To be clear, we will set up *.vm support via both these mechanisms if both happen to be present? I believe we discussed that being safe and the approach, but wanted to confirm.